argocd/datadog_checks/argocd/check.py

```diff
@@ -0,0 +1,108 @@
+                "`app_controller_endpoint`, `repo_server_endpoint` or `api_server_endpoint`."
+            )
+
+        if app_controller_endpoint:
+            self.scraper_configs.append(
+                self._generate_config(app_controller_endpoint, APP_CONTROLLER_NAMESPACE, APPLICATION_CONTROLLER_METRICS)
+            )
+        if api_server_endpoint:
+            self.scraper_configs.append(
+                self._generate_config(api_server_endpoint, API_SERVER_NAMESPACE, API_SERVER_METRICS)
+            )
+        if repo_server_endpoint:
+            self.scraper_configs.append(
+                self._generate_config(repo_server_endpoint, REPO_SERVER_NAMESPACE, REPO_SERVER_METRICS)
+            )
+
+    def _generate_config(self, endpoint, namespace, metrics):
+        config = {
+            'openmetrics_endpoint': endpoint,
+            'metrics': metrics,
+            'namespace': namespace,
+        }
+        config.update(self.instance)
+        return config
+
+    def configure_transformer_go_memstats_alloc_bytes(self, metric_name):
+        # Custom transformer to ensure we only pick up the gauge version of this metric
+        # ArgoCD exposes a similar metric `go_memstats_alloc_bytes_total` as a counter
+        # which would also be collected under the same name and could cause inaccuracies
+        # in the metric value
+        def go_memstats_alloc_bytes_transformer(metric, sample_data, runtime_data):
+            for sample, tags, hostname in sample_data:
+                self.gauge(metric_name, sample.value, tags=tags, hostname=hostname)
+
+        return go_memstats_alloc_bytes_transformer
+
+    def configure_transformer_argocd_cluster_connection_status(self, metric_name):
+        # The metric reports a 1 if connected, and 0 if not. The mapping used here:
+        # OK if connected || OK if metric value is `1`
+        # Critical if not connected || Critical if value is '0'
+        # Unknown for everything else
+        status_map = {1: ServiceCheck.OK, 0: ServiceCheck.CRITICAL}
+        service_check_method = self.service_check
+
+        def argocd_cluster_connection_status_transformer(metric, sample_data, runtime_data):
+            for sample, _, hostname in sample_data:
+                service_check_method(
+                    metric_name,
+                    status_map.get(int(sample.value), ServiceCheck.UNKNOWN),
+                    hostname=hostname,
+                )
+
+        return argocd_cluster_connection_status_transformer
+
+    def configure_additional_transformers(self):
+        if not self.scrapers:
+            return
+        for endpoint in self.instance.keys():
+            if endpoint == "app_controller_endpoint":
+                self.scrapers[self.instance[endpoint]].metric_transformer.add_custom_transformer(
+                    ("argocd_cluster_connection_status"),
+                    self.configure_transformer_argocd_cluster_connection_status("cluster.connection.status"),
```

Comment:
```suggestion
                    self.configure_transformer_argocd_cluster_connection_status("argocd.app_controller.cluster.connection.status"),
```

Reply:
The namespace is added depending the endpoint used:
https://github.com/DataDog/integrations-core/blob/704425e2018e2463026d817f78389104debed4fd/argocd/datadog_checks/argocd/check.py#L46